File: client/java/src/avt/Effect.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public abstract class Effect {
   public boolean isStop = false;
   public short IDAction = -1;

   public abstract void updateWind();

   public abstract void paint(Graphics var1);

   public final void show() {
      Canvas.currentEffect.addElement(this);
   }

   public void close() {
      Canvas.currentEffect.removeElement(this);
   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/Effect.cs
public abstract class Effect
{
	public bool isStop;

	public short IDAction = -1;

	public abstract void update();

	public abstract void paint(MyGraphics g);

	public void show()
	{
		Canvas.currentEffect.addElement(this);
	}

	public virtual void close()
	{
		Canvas.currentEffect.removeElement(this);
	}

	public virtual void paintBack(MyGraphics g)
	{
	}
}
